notebooks/01-Summary-statistics-and-plots.ipynb

```diff
@@ -0,0 +1,265 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def scale_df(df: pd.DataFrame, scale:float, image:str) -> pd.DataFrame:\n",
+    "    \"\"\"Scale the numerical values of a data frame. Retains string variables and the index.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    df: pd.DataFrame\n",
+    "        Pandas Dataframe to scale.\n",
+    "    scale: float\n",
+    "        Factor by which to scale the data.\n",
+    "    image: str\n",
+    "        Name for new (dummy) image.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    pd.DataFrame\n",
+    "        Scaled data frame\n",
+    "    \"\"\"\n",
+    "    _df = df[df.select_dtypes(include=['number']).columns] * scale\n",
+    "    _df.reset_index(inplace=True)\n",
+    "    _df[\"image\"] = image\n",
+    "    _df = pd.concat([_df, df[[\"circular\", \"basename\"]]], axis=1)\n",
+    "    _df.set_index([\"image\", \"threshold\", \"molecule_number\"], inplace=True)\n",
+    "    return _df\n",
+    "smaller = scale_df(df, scale=0.9, image=\"smaller\")\n",
+    "larger = scale_df(df, scale=1.25, image=\"larger\")\n",
+    "df_three_images = pd.concat([smaller, df, larger])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6a57eb1a-3093-4ca6-8411-e62ea2614730",
+   "metadata": {},
+   "source": [
+    "### Contour Length from Three Processed Images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9820d259-680e-49fd-a1eb-e14d200a0b7a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_three_images[\"contour_lengths\"].groupby(\"image\").plot.hist(figsize=(16,9),\n",
+    "                                bins=20,\n",
+    "                                title=\"Contour Lengths\",\n",
+    "                                alpha=0.5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e1bdba05-e0a6-42cc-bddd-ed59fe3d45f5",
+   "metadata": {},
+   "source": [
+    "### Violin Plot of `max_feret` using Seaborn\n",
+    "\n",
+    "Pandas does not have built-in support for Violin Plots so we switch to using Seaborn."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "68aa386c-fa7a-4ee3-8310-83e4b62d67a9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Reset dataframe index to make `image` readily available\n",
+    "df_three_images.reset_index(inplace=True)\n",
+    "fig, ax = plt.subplots(1, 1, figsize=(16,9))\n",
+    "sns.violinplot(data=df_three_images, x=\"image\", y=\"max_feret\", hue=\"image\", alpha=0.5)\n",
+    "plt.title(\"Minimum Feret\")\n",
+    "plt.ylabel(\"Minimum Feret / nm\")\n",
+    "# Return the index\n",
+    "df_three_images.set_index([\"image\", \"threshold\", \"molecule_number\"], inplace=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3ae80090-71d3-4c87-9907-fd10b892c962",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_three_images.index\n"
```

Comment:
Is this code block intended to be left in? I'm not sure why it would be printed unless you want to highlight something about the dataframe? Apologies if I'm misunderstanding, just wanted to make sure.

It prints as this:

```
MultiIndex([('smaller', 'upper', 0.0),
            ('smaller', 'upper', 1.0),
            ('smaller', 'upper', 2.0),
            ('smaller', 'upper', 3.0),
            ('smaller', 'upper', 4.0),
            ('smaller', 'upper', 5.0),
            ('smaller', 'upper', 6.0),
            ('smaller', 'upper', 7.0),
            ('smaller', 'upper', 8.0),
            ('smaller', 'upper', 9.0),
            ...
            (      nan,     nan, nan),
            (      nan,     nan, nan),
            (      nan,     nan, nan),
            (      nan,     nan, nan),
            (      nan,     nan, nan),
            (      nan,     nan, nan),
            (      nan,     nan, nan),
            (      nan,     nan, nan),
            (      nan,     nan, nan),
            (      nan,     nan, nan)],
           names=['image', 'threshold', 'molecule_number'], length=105)
```